swift-tools-support-core/Sources/TSCBasic/FileSystem.swift

```diff
@@ -11,74 +11,81 @@
 import TSCLibc
 import Foundation
 
-public enum FileSystemError: Swift.Error {
+public enum FileSystemError: Swift.Error, Equatable {
     /// Access to the path is denied.
     ///
     /// This is used when an operation cannot be completed because a component of
     /// the path cannot be accessed.
     ///
     /// Used in situations that correspond to the POSIX EACCES error code.
-    case invalidAccess
+    case invalidAccess(AbsolutePath)
 
     /// IO Error encoding
     ///
     /// This is used when an operation cannot be completed due to an otherwise
     /// unspecified IO error.
-    case ioError
+    case ioError(AbsolutePath)
```

Comment:
If we're passing something here, shouldn't we be passing the IO error code instead?